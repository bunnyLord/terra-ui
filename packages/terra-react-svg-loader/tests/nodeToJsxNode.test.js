import nodeToJsxNode from '../src/nodeToJsxNode';

describe('nodeToJsxNode', () => {
  it('white-listed attributes should change to jsx', () => {
    const input = '<path fill-rule="evenodd" />';
    document.body.innerHTML = input;
    expect(nodeToJsxNode(document.body.children[0])).toBe('<path fillRule="evenodd" ></path>');
    expect(nodeToJsxNode(document.body.children[0])).toMatchSnapshot();
  });
  it('non-hyphenated attributes should stay the same', () => {
    const input = '<path fill="#232" />';
    document.body.innerHTML = input;
    expect(nodeToJsxNode(document.body.children[0])).toBe('<path fill="#232" ></path>');
    expect(nodeToJsxNode(document.body.children[0])).toMatchSnapshot();
  });
  it('attribute class should change to className', () => {
    const input = '<path class="math" />';
    document.body.innerHTML = input;
    expect(nodeToJsxNode(document.body.children[0])).toBe('<path className="math" ></path>');
    expect(nodeToJsxNode(document.body.children[0])).toMatchSnapshot();
  });
});
