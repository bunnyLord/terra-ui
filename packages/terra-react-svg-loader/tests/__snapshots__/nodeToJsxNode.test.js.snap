exports[`nodeToJsxNode attribute class should change to className 1`] = `"<path className=\"math\" ></path>"`;

exports[`nodeToJsxNode non-hyphenated attributes should stay the same 1`] = `"<path fill=\"#232\" ></path>"`;

exports[`nodeToJsxNode white-listed attributes should change to jsx 1`] = `"<path fillRule=\"evenodd\" ></path>"`;
